openedx_lti_tool_plugin/tests/test_views.py

```diff
@@ -609,6 +694,99 @@ def test_post_enroll_raises_exception(
+    @patch.object(CourseKey, 'from_string', return_value='random-course-key')
+    @patch.object(DjangoMessageLaunch, 'get_launch_data', return_value=BASE_LAUNCH_DATA)
+    @patch('openedx_lti_tool_plugin.views.DjangoDbToolConf')
+    @patch.object(DjangoMessageLaunch, '__init__', return_value=None)
+    def test_post_with_allow_complete_course_launch_disabled(
+        self,
+        message_launch_mock: MagicMock,  # pylint: disable=unused-argument
+        tool_conf_mock: MagicMock,  # pylint: disable=unused-argument
+        get_launch_data_mock: MagicMock,  # pylint: disable=unused-argument
+        course_key_mock: MagicMock,  # pylint: disable=unused-argument
+        authenticate_and_login_mock: MagicMock,  # pylint: disable=unused-argument
+        enroll_mock: MagicMock,  # pylint: disable=unused-argument
+        allow_complete_course_launch_mock: MagicMock,
+        log,
+    ):
+        """Test POST request with correct Course ID and Usage Key.
+
+        Args:
+            usage_key_mock: Mocked UsageKey class.
+            message_launch_mock: Mocked DjangoMessageLaunch class.
+            get_launch_data_mock: Mocked DjangoMessageLaunch get_launch_data method.
+            course_key_mock: Mocked CourseKey from_string method.
+            authenticate_and_login_mock: Mocked LtiToolLaunchView authenticate_and_login method.
+            enroll_mock: Mocked LtiToolLaunchView enroll method.
+            allow_complete_course_launch_mock: Mocked allow_complete_course_launch waffle switch.
+            log: LogCapture fixture.
+        """
+        allow_complete_course_launch_mock.is_enabled.return_value = False
+        request = self.factory.post(self.url)
+        request.user = self.user
+
+        response = self.view_class.as_view()(request, COURSE_ID)
+
+        allow_complete_course_launch_mock.is_enabled.assert_called_once_with()
+        log.check(
+            (
+                'openedx_lti_tool_plugin.views',
+                'ERROR',
+                'LTI 1.3: Complete course launches are not enabled.',
+            ),
+        )
+        self.assertEqual(response.content.decode('utf-8'), self.view_class.BAD_RESPONSE_MESSAGE)
+        self.assertEqual(response.status_code, 400)
+
+    @patch(
+        'openedx_lti_tool_plugin.views.redirect',
+        return_value=MagicMock(status_code=200, content='random-content'),
+    )
+    @patch('openedx_lti_tool_plugin.views.ALLOW_COMPLETE_COURSE_LAUNCH')
+    @patch.object(LtiToolLaunchView, 'enroll')
+    @patch.object(LtiToolLaunchView, 'authenticate_and_login', return_value='random-user')
+    @patch.object(CourseKey, 'from_string', return_value='random-course-key')
+    @patch.object(DjangoMessageLaunch, 'get_launch_data', return_value=BASE_LAUNCH_DATA)
+    @patch('openedx_lti_tool_plugin.views.DjangoDbToolConf')
+    @patch.object(DjangoMessageLaunch, '__init__', return_value=None)
+    def test_post_with_allow_complete_course_launch_enabled(
```

Comment:
```suggestion
    def test_post_with_allow_complete_course_launch_enabled(
```
@alexjmpb Aren't we already covering this on the first test?